modin/numpy/arr.py

```diff
@@ -1469,6 +1559,249 @@ def sum(
                 + initial
             )
 
+    def all(self, axis=None, out=None, keepdims=None, *, where=True):
+        check_kwargs(keepdims=keepdims, where=where)
+        if self._ndim == 1:
+            if axis == 1:
+                raise numpy.AxisError(1, 1)
+            result = self._query_compiler.all(axis=0)
+            if keepdims:
+                if out is not None and out.shape != (1,):
+                    raise ValueError(
+                        f"operand was set up as a reduction along axis 0, but the length of the axis is {out.shape[0]} (it has to be 1)"
+                    )
+                return fix_dtypes_and_determine_return(result, 1, bool, out, where)
+            result = result.to_numpy()[0, 0]
+            return result if not where else result and where
+        if axis is None:
+            target = where.where(self, True) if isinstance(where, array) else self
+            result = target.all(axis=1, keepdims=None).all(
```

Comment:
We can just use `logical_and` on the query compiler twice (once along each axis).

Reply:
I'm not quite sure what you mean here, since `logical_and` doesn't do any reductions.